src/lib.rs

```diff
@@ -16,13 +16,26 @@ extern crate derivative;
 extern crate bench_utils;
 
 use ark_ff::Field;
+use core::iter::FromIterator;
 pub use ark_poly::DensePolynomial as Polynomial;
 use rand_core::RngCore;
 
-use ark_std::{
-    borrow::Cow,
+#[cfg(not(feature = "std"))]
+#[macro_use]
+extern crate alloc;
+
+#[cfg(not(feature = "std"))]
+use alloc::{
+    collections::{BTreeMap, BTreeSet},
+    rc::Rc,
+    string::{String, ToString},
+    vec::Vec,
+};
+
+#[cfg(feature = "std")]
+use std::{
     collections::{BTreeMap, BTreeSet},
-    iter::FromIterator,
+    rc::Rc,
     string::{String, ToString},
     vec::Vec,
 };
```

Comment:
This change slipped in incorrectly; we want to just import things from `ark_std`